### DISCUSSION TITLE
Hydration not working on class attribute?

### DISCUSSION THREAD

**Discussion by some-user123** [nuxt/nuxt#16827] [Questions]
Not sure, whether this is a feature or a bug and whether it is nuxt or vue or I don't know...

I'm storing information in localStorage (+ using pinia). So, apparently the state on server and client is different.

The state is accessed in a computed property `const computedClass = computed(() => ...)`.

Rendered in the template (`{{ computedClass }}`) the server value is shown directly after loading the (server rendered) page and quickly switches to the client value. That's good.

However, if I use the same computed property as a class attribute `<span :class="computedClass">...`, it does not get updated with the client value but remains on the server value. That's bad. How can I change that?

Reproduction: 
```html
<template>
  <div>
    {{ computedClass }} <br />
    <span :class="computedClass"></span>
  </div>
</template>

<script setup>
function someData() {
  // will return true on client, but false on server
  //
  // simplified here ;-)
  // actually involves pinia and localStorage and data in local storage
  return process.client;
}

const computedClass = computed(() => (someData() ? 'client' : 'server'));
</script>

<style>
.client::after {
  content: 'client';
}

.server::after {
  content: 'server';
}
</style>
```
https://stackblitz.com/edit/nuxt-starter-5nkgd6?file=app.vue

The result is inconsistent as it's mixing client and server values.

I know, I can wrap things in `<client-only>` to disable SSR and that would solve the issue here. However, this isn't really an option, as it would disable SSR for the whole app because I'm wrapping my whole app in an element with `:class="computedClass"`.

**Comment by TheAlexLichter** [comment]
Random idea: Instead of a `computed`, use a `ref` + change the class in `onBeforeMount`. This could avoid state mismatch

**Reply by some-user123** [reply]
You mean like this?
```javascript
const computedClass = ref('');
onBeforeMount(() => {
  computedClass.value = someData() ? 'client' : 'server';
});
```
https://stackblitz.com/edit/nuxt-starter-ruewp7?file=app.vue

Doesn't work :-(  The client does not overwrite the server rendered class attribute...

**Comment by danielroe** [comment]
You should not try to change the class before the initial mount. Normally this would be a hydration error, but in the case of class + style attributes, Vue doesn't error, but it won't update them.

**Reply by some-user123** [reply]
In other words, you would suggest:
```javascript
const computedClass = ref('');
onMounted(() => {
  computedClass.value = someData() ? 'client' : 'server';
});
```
Correct? That seems to work...

**Reply by danielroe** [reply]
Yes, perfect 👍 

Though mounted will never be called on server, fyi. You can use `useState` to get around this:

```js
const computedClass = useState(() => someData() ? 'client' : 'server');
onMounted(() => {
  computedClass.value = 'client';
});
```
